src/draw.c

```diff
@@ -738,37 +835,38 @@ static void render_content(cairo_t *c, struct colored_layout *cl, int width, dou
                              frame_x = width - progress_width - settings.h_padding;
                              break;
                 }
-                unsigned int x_bar_1 = frame_x + frame_width,
-                             x_bar_2 = x_bar_1 + progress_width_1;
 
-                double half_frame_width = frame_width / 2.0;
+                unsigned int x_bar = frame_x + frame_width;
+                float half_frame_width = (float)frame_width / 2.0f;
 
                 /* Draw progress bar
                 * TODO: Modify draw_rounded_rect to fix blurry lines due to fractional scaling
-                * Note: the bar could be drawn a bit smaller, because the frame is drawn on top 
+                * Note: the bar could be drawn a bit smaller, because the frame is drawn on top
                 */
-                // left side (fill)
-                cairo_set_source_rgba(c, cl->highlight.r, cl->highlight.g, cl->highlight.b, cl->highlight.a);
-                draw_rounded_rect(c, x_bar_1, frame_y, progress_width_1, progress_height, 
-                        settings.progress_bar_corner_radius, scale, true, true);
-                cairo_fill(c);
+
+                // progress_bar_corners
                 // right side (background)
                 cairo_set_source_rgba(c, cl->bg.r, cl->bg.g, cl->bg.b, cl->bg.a);
-                draw_rounded_rect(c, x_bar_2, frame_y, progress_width_2, progress_height, 
-                        settings.progress_bar_corner_radius, scale, true, true);
+                draw_rounded_rect(c, x_bar, frame_y, progress_width_without_frame, progress_height,
+                        settings.progress_bar_corner_radius, scale, settings.corners);
+                cairo_fill(c);
 
+                // left side (fill)
+                cairo_set_source_rgba(c, cl->highlight.r, cl->highlight.g, cl->highlight.b, cl->highlight.a);
+                draw_rounded_rect(c, x_bar, frame_y, progress_width_fill, progress_height,
+                        settings.progress_bar_corner_radius, scale, settings.corners);
                 cairo_fill(c);
 
                 // border
                 cairo_set_source_rgba(c, cl->frame.r, cl->frame.g, cl->frame.b, cl->frame.a);
                 cairo_set_line_width(c, frame_width * scale);
                 draw_rounded_rect(c,
                                 frame_x + half_frame_width,
-                                frame_y + half_frame_width,
```

Comment:
This seems to improve/fix one of the problem in #1178 where the bar frame is offset from the fill